src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,191 @@ static DamageTable D_809901F8 = {
+ */
+void EnTg_Idle(EnTg* this, PlayState* play) {
+    Vec3f heartStartPos;
+
+    this->actor.shape.rot.y += sREG(0) + 0x258;
+    this->actor.world.rot = this->actor.shape.rot;
+
+    if (DECR(this->spawnHeartTimer) == 0) {
+        this->spawnHeartTimer = 12;
+        heartStartPos = this->actor.world.pos;
+        heartStartPos.y += 62.0f;
+        EnTg_SpawnHeart(this, this->effects, &heartStartPos, ARRAY_COUNT(this->effects));
+    }
+}
+
+void EnTg_Update(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnTg_UpdateSkelAnime(this, play);
+    EnTg_UpdateHearts(play, this->effects, ARRAY_COUNT(this->effects));
+    EnTg_UpdateCollider(this, play);
+}
+
+s32 EnTg_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return false;
+}
+
+void EnTg_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTg* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+
+    if (limbIndex == HONEY_AND_DARLING_LIMB_MAN_HEAD) {
+        Matrix_MultVec3f(&zeroVec, &this->actor.focus.pos);
+    }
+}
+
+void EnTg_Draw(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+    GraphicsContext* gfxCtx;
+
+    Matrix_Push();
+    EnTg_DrawHeart(play, this->effects, ARRAY_COUNT(this->effects));
+    Matrix_Pop();
+
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Gfx_EnvColor(play->state.gfxCtx, 0, 50, 160, 0));
+    gSPSegment(POLY_OPA_DISP++, 0x09, Gfx_EnvColor(play->state.gfxCtx, 255, 255, 255, 0));
+
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnTg_OverrideLimbDraw, EnTg_PostLimbDraw, &this->actor);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+/**
+ * Spawns a heart at the first effects array index that's not enabled.
+ * Because of the frame counts, only two hearts are ever spawned at a time.
+ */
+void EnTg_SpawnHeart(EnTg* this, EnTgHeartEffect* effects, Vec3f* heartStartPos, s32 len) {
+    Vec3f heartVelocityVec = { 0.0f, 1.5f, 0.0f };
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+
+    for (i = 0; i < len && effects->isEnabled; i++, effects++) {}
+
+    if (i < len) {
+        effects->isEnabled = true;
+        effects->pos = *heartStartPos;
+        effects->velocity = heartVelocityVec;
+        effects->unusedZeroVec = zeroVec;
+        effects->scale = 0.01f;
+        effects->pos.x += 4.0f * Math_SinS(this->actor.shape.rot.y);
+        effects->pos.z += 4.0f * Math_CosS(this->actor.shape.rot.y);
+        effects->timer = 16;
+    }
+}
+
+/**
+ * The heart path is curvy as it floats up because of the use of Math_SinS and Math_CosS.
+ */
+void EnTg_UpdateHearts(PlayState* play, EnTgHeartEffect* effects, s32 len) {
```

Comment:
```suggestion
void EnTg_UpdateHearts(PlayState* play, EnTgHeartEffect* effect, s32 numEffects) {
```
I don't think `effect` needs the plural since it only points to 1 at a time.
Also, `len` isn't used for total very much, something like `numEffects` is more to our style